be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -522,6 +522,13 @@ Status SegmentIterator::_lookup_ordinal_from_pk_index(const RowCursor& key, bool
     std::string index_key;
     encode_key_with_padding<RowCursor, true, true>(
             &index_key, key, _segment->_tablet_schema->num_key_columns(), is_include);
+    if (index_key <= _segment->min_key()) {
```

Comment:
use `<` and `>` instead, notice L549, if `index_key` is contained in our index, we should process parameter `is_include` as well.